sdcm/utils/version_utils.py

```diff
@@ -197,3 +220,83 @@ def get_scylla_docker_repo_from_version(version: str):
     if is_enterprise(version):
         return 'scylladb/scylla-enterprise'
     return 'scylladb/scylla'
+
+
+def resolve_latest_repo_symlink(url: str) -> str:
+    """Resolve `url' to the actual latest repo link if it contains `latest' substring, otherwise, return `url' as is.
+
+    If `url' doesn't point to the latest repo file then raise critical ScyllaRepoEvent and return `url' as is.
+
+    Can raise ValueError if `url' is not a valid URL that points to a repo file stored in S3.
+    Or can raise RuntimeError if it's something wrong with S3 directory.
+    """
+    base, sep, rest = url.partition("latest")
+    if sep != "latest":
+        LOGGER.info("%s doesn't contain `latest' substring, use URL as is", url)
+        return url
+
+    parsed_base_url = urlparse(base)
+
+    # URL can be in 3 forms:
+    #  1. http://downloads.scylladb.com/...
+    #  2. http://downloads.scylladb.com.s3.amazonaws.com/...
+    #  3. http://s3.amazonaws.com/downloads.scylladb.com/...
+    # Plus same forms for HTTPS.
+    if parsed_base_url.netloc in (SCYLLA_REPO_BUCKET, f"{SCYLLA_REPO_BUCKET}.s3.amazonaws.com", ):
+        prefix = parsed_base_url.path.lstrip("/")
+    elif parsed_base_url.netloc == "s3.amazonaws.com" and parsed_base_url.path.startswith(f"/{SCYLLA_REPO_BUCKET}/"):
+        prefix = parsed_base_url.path.split("/", 2)[-1]
+    else:
+        raise ValueError(f"Unsupported URL: {url}")
+
+    s3_client = boto3.client("s3", region_name=DEFAULT_AWS_REGION)
```

Comment:
maybe we should start doing all those anonymously ? since this bucket is public. 
https://stackoverflow.com/questions/34865927/can-i-use-boto3-anonymously

Reply:
Good point. Changes was made.